bl31/aarch64/crash_reporting.S

```diff
@@ -327,18 +327,16 @@ func do_crash_reporting
 
 	bl	plat_crash_console_flush
 
-	/* Done reporting */
-	no_ret	plat_panic_handler
+crash_panic:
+	/* Done reporting, restore original x30 */
+	mrs	x0, tpidr_el3
+	ldr	x30, [x0, #REG_SIZE * 7]
```

Comment:
At this point `[x0, #REG_SIZE * 7]` no longer contains x30 as this location has been re-used for other purposes (e.g. in line 231). It's not going to be easy to restore x30 in this case so I suggest leaving it as it was.